### PR TITLE
[COST-5382] konflux: update bundle pipeline and use brew registry

### PR DESCRIPTION
**Pull request by djnakabaale**
- update operator bundle pipeline to only run when bundle assets are updated

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Build the manager binary
-FROM --platform=${BUILDPLATFORM:-linux/amd64} docker.io/library/golang:1.22.6 AS builder
+FROM --platform=${BUILDPLATFORM:-linux/amd64} brew.registry.redhat.io/rh-osbs/openshift-golang-builder:v1.22 AS builder
```
Comment (samdoran) [inline Dockerfile:2]: The base image for upstream should not change. This should be updated during release.

File: .tekton/koku-metrics-operator-bundle-pull-request.yaml [hunk 2]
```diff
@@ -7,8 +7,13 @@ metadata:
...
     pipelinesascode.tekton.dev/max-keep-runs: "3"
-    pipelinesascode.tekton.dev/on-cel-expression: event == "pull_request" && target_branch
-      == "main"
+    pipelinesascode.tekton.dev/on-cel-expression: event == "pull" && target_branch == "main" && 
+      (
+        ".tekton/koku-metrics-operator-bundle-pull.yaml".pathChanged() || 
+        "bundle/metadata/***".pathChanged() || 
+        "bundle/manifests/***".pathChanged() || 
+        "bundle.Dockerfile".pathChanged() 
+      )
```
Comment (samdoran) [inline .tekton/koku-metrics-operator-bundle-pull-request.yaml:16]: What's the purpose of changing the trigger for the pipeline? I usually see this used when setting up nudges, but I don't think we have components that need nudging.

File: Dockerfile (hunk 1 above)
Comment (samdoran) [reply Dockerfile:2]: Everyone seems to solve this differently. We could maintain separate containerfiles for Konfux and use that in the pipeline params, or we could use `BUILD_ARGS` to pass in a variable that sets the base image.

File: .tekton/koku-metrics-operator-bundle-pull-request.yaml (hunk 2 above)
Comment (djnakabaale) [reply .tekton/koku-metrics-operator-bundle-pull-request.yaml:16]: This pipeline doesn't need to run every time but only when bundle assets are updated. In the cluster service yaml in the manifests folder, we reference the [operator image](https://github.com/project-koku/koku-metrics-operator/blob/0ec274a9e1c7f3b581018dbf29870348156761a1/bundle/manifests/koku-metrics-operator.clusterserviceversion.yaml#L42) which will be replaced with the konflux-built image and once everything is configured, that's what will be nudged.

File: Dockerfile (hunk 1 above)
Comment (djnakabaale) [reply Dockerfile]: Right, I think we can leverage separate [dockerfiles in the pipeline params](https://github.com/project-koku/koku-metrics-operator/blob/0ec274a9e1c7f3b581018dbf29870348156761a1/.tekton/koku-metrics-operator-pull-request.yaml#L29).  But we can revisit this once we have downstream builds.

**Review by samdoran — APPROVED** [review]

**Pull request merged**